extensions/git/src/repository.ts

```diff
@@ -1794,10 +1794,21 @@ export class Repository implements Disposable {
 		return this.pullFrom(false, remote, branch, unshallow);
 	}
 
+	private getAutoStashConfig(config: WorkspaceConfiguration) {
```

Comment:
I would prefer if we would do this differently. We should update `maybeAutoStash` so that `shouldAutoStash` factors in the git version. If the git version does support the `--autostash` flag, then `shouldAutoStash` should be `false`. Then read the setting, and pass it down through the `options`.